pkg/server/endpoints/registration/handler.go

```diff
@@ -545,6 +547,95 @@ func (h *Handler) ListAgents(ctx context.Context, listReq *registration.ListAgen
 	return &registration.ListAgentsResponse{Nodes: resp.Nodes}, nil
 }
 
+func (h *Handler) MintX509SVID(ctx context.Context, req *registration.MintX509SVIDRequest) (_ *registration.MintX509SVIDResponse, err error) {
+	counter := telemetry_registrationapi.StartMintX509SVIDCall(h.Metrics)
+	addCallerIDLabel(ctx, counter)
+	defer counter.Done(&err)
+
+	if req.SpiffeId == "" {
+		return nil, status.Error(codes.InvalidArgument, "request missing SPIFFE ID")
+	}
+	if len(req.Csr) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "request missing CSR")
+	}
+
+	spiffeID, err := idutil.NormalizeSpiffeID(req.SpiffeId, idutil.AllowTrustDomainWorkload(h.TrustDomain.Host))
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	csr, err := x509.ParseCertificateRequest(req.Csr)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: %v", err)
+	}
+	if err := csr.CheckSignature(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: signature verify failed")
+	}
+
+	svid, err := h.ServerCA.SignX509SVID(ctx, ca.X509SVIDParams{
+		SpiffeID:  spiffeID,
+		PublicKey: csr.PublicKey,
+		TTL:       time.Duration(req.Ttl) * time.Second,
+		DNSList:   req.DnsNames,
```

Comment:
should we call `validateDNS()` on these dns names?

Reply:
good call